clvm/core_ops.py

```diff
@@ -21,46 +28,48 @@ def op_if(args: SExp):
     return IF_COST, r.first()
 
 
-def op_cons(args: SExp):
+def op_cons(args: _T_SExp) -> Tuple[int, _T_SExp]:
     if args.list_len() != 2:
         raise EvalError("c takes exactly 2 arguments", args)
     return CONS_COST, args.first().cons(args.rest().first())
 
 
-def op_first(args: SExp):
+def op_first(args: _T_SExp) -> Tuple[int, _T_SExp]:
     if args.list_len() != 1:
         raise EvalError("f takes exactly 1 argument", args)
     return FIRST_COST, args.first().first()
 
 
-def op_rest(args: SExp):
+def op_rest(args: _T_SExp) -> Tuple[int, _T_SExp]:
     if args.list_len() != 1:
         raise EvalError("r takes exactly 1 argument", args)
     return REST_COST, args.first().rest()
 
 
-def op_listp(args: SExp):
+def op_listp(args: _T_SExp) -> Tuple[int, SExp]:
```

Comment:
Look this over and also check other adjacent cases.

```suggestion
def op_listp(args: SExp) -> Tuple[int, SExp]:
```